src/graphics/text.rs

```diff
@@ -271,41 +274,49 @@ impl Drawable for FontAtlas {
     }
 }
 
+const DEFAULT_TEXT_BUFFER_SIZE: usize = 64;
+
 #[derive(Debug)]
 pub struct Text {
     batch: SpriteBatch,
-    char_texture_map: HashMap<char, CharInfo>,
+    atlas: Cached<FontAtlas>,
 }
 
 impl Text {
-    pub fn new(ctx: &mut Graphics, input_text: &str, font_atlas: &FontAtlas, color: Color) -> Self {
-        let mut text = Text {
+    pub fn new(ctx: &mut Graphics, input_text: &str, font_atlas: FontAtlas, color: Color) -> Self {
```

Comment:
Wouldn't passing FontAtlas here instead of &FontAtlas result in a move, thereby making it impossible to make any other Text objects?

Reply:
`FontAtlas` is `Clone`, so no.

Reply:
Yeah I know it's `Clone`, it seems wasteful to clone the whole thing unless you're tossing this into some sort of reference counter.

Reply:
well, yes. it's getting tossed into `Cached`.